staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/validating/dispatcher.go

```diff
@@ -197,6 +196,11 @@ func (d *validatingDispatcher) Dispatch(ctx context.Context, attr admission.Attr
 			}
 
 			if callErr, ok := err.(*webhookutil.ErrCallingWebhook); ok {
+				if errors.Is(callErr.Reason, context.Canceled) {
+					klog.Warningf("Context Canceled when calling webhook %v", hook.Name)
+					errCh <- apierrors.NewInternalError(err)
+					return
+				}
 				if ignoreClientCallFailures {
```

Comment:
so here just doing 

```
errCh <- apierrors.NewInternalError(err)
return
```
is not necessarily the same behavior. The `if ignoreClientCallFailures` block uses utilruntime.HandlerError while the other block does the `errCh <- apierrors.NewInternalError(err)`

Reply:
I think it probably makes sense to always do `errCh <- apierrors.NewInternalError(err)` for context cancelled but will differ to @liggitt